extension/js/common/settings.ts

```diff
@@ -276,6 +276,7 @@ export class Settings {
           window.location.href = Url.create('/chrome/settings/setup.htm', { acctEmail: response.acctEmail, idToken: response.id_token });
         }
       } else if (response.result === 'Denied' || response.result === 'Closed') {
+        console.log({ response, settingsTabId })
         if (settingsTabId) {
           await Settings.renderSubPage(acctEmail, settingsTabId, '/chrome/settings/modules/auth_denied.htm');
```

Comment:
@tomholub Here, how do I check if the subPage is already rendered? ATM it renders another iframe inside the existing iframe and so on.

Reply:
I think it's supposed to replace whatever subPage is currently rendered. If it renders subpage inside a subpage, then maybe `newGoogleAcctAuthPromptThenAlertOrForward` is currently getting called from some subpage (just guessing)? In which case the caller may be errorneously giving `newGoogleAcctAuthPromptThenAlertOrForward` it's own `tabId` instead of the `tabId` of the settings page, which from the perspective of the calling subpage would be `parentTabId`.

Then this calls the parent, which is the settings page, which will render a subpage which will replace whatever subpage called it.

NOW, to get out of this whole mess another way, we could consider rendering a modal instead of a subpage. The modal can have the same information and same options. Problem solved, it's easier to render a modal in all sorts of context than it's to switch to a subpage.

Reply:
Is there existing functionality for rendering a modal? Can't find it.

Reply:
You can `await Ui.modal.warning`

Reply:
It actually uses a library made by @limonte here

Reply:
Yes, I saw this, but it just renders plain text inside the modal while we need it to render HTML with custom functionality bound to buttons:

![image](https://user-images.githubusercontent.com/2179726/96570928-9b6ddd00-12d3-11eb-8157-8ffda1010591.png)

Something like `await Ui.modal.warning('/chrome/settings/modules/auth_denied.htm')` would be great, but it obviously doesn't support this.

Reply:
Indeed we don't have that yet -  you can add a method like `Ui.modal.html` that will be like `info`, but accepts html. Make sure to use the Xss class to sanitize the html. It will be useful for other stuff too I'm sure.

Reply:
I see `error` has `isHtml` param - you can do the same for info

Reply:
Please, have a look now. The only thing we don't get the "Update permissions" button in the modal, because the scripts get sanitized.

Reply:
Looking at the page, the whole thing can get simplified. You can throw out all of the auth_denied javascript and html file. Use `Ui.modal.confirm` but update the method to also accept html using `isHtml` param. Then use the text as above with the link(make sure the link opens in a new tab), but at the end add a question: `Would you like to proceed to the permissions screen again?` Which then works nicely with the modal options: ok | cancel. 

That way you can throw away a bunch of useless code, which is always a win.

Reply:
That means you don't need any js inside the modal, you just await the user response (true or false) and based on that initiate again or not.

Reply:
Done, please check.